src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -1221,6 +1221,12 @@ private void AddErrorMappingToExecutorMethod(OpenApiUrlTreeNode currentNode, Ope
                     var obsoleteFactoryMethod = (CodeMethod)originalFactoryMethod.Clone();
                     obsoleteFactoryMethod.ReturnType = new CodeType { Name = obsoleteTypeName, TypeDefinition = obsoleteClassDefinition };
                     obsoleteClassDefinition.AddMethod(obsoleteFactoryMethod);
+                    var originalSerializerMethod = codeClass.Methods.First(static x => x.Kind is CodeMethodKind.Serializer);
+                    var obsoleteSerializerMethod = (CodeMethod)originalSerializerMethod.Clone();
+                    obsoleteClassDefinition.AddMethod(obsoleteSerializerMethod);
+                    var originalDeserializerMethod = codeClass.Methods.First(static x => x.Kind is CodeMethodKind.Deserializer);
+                    var obsoleteDeserializerMethod = (CodeMethod)originalDeserializerMethod.Clone();
+                    obsoleteClassDefinition.AddMethod(obsoleteDeserializerMethod);
```

Comment:
One of the tests was throwing an exception, @rkodev if you should be able to see it if you revert, however, its a subtle bug since to reproduce you have to use the same arguments the test uses, which normally isn't the case